src/test/helpers/log_test.h

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include <gtest/gtest.h>
+
+#include <QDebug>
+#include <QList>
+#include <QRegularExpression>
+#include <tuple>
+
+#define SETUP_LOG_CAPTURE() \
+    qInstallMessageHandler(logCapture)
+
+#define ASSERT_ALL_EXPECTED_MSG()                                                                  \
+    if (!logMessagesExpected.isEmpty()) {                                                          \
+        QString errMsg;                                                                            \
+        QDebug strm(&errMsg);                                                                      \
+        strm << logMessagesExpected.size() << "expected log messages didn't occur: \n";            \
+        for (const auto& msg : std::as_const(logMessagesExpected))                                 \
+            strm << "\t" << std::get<QtMsgType>(msg) << std::get<QRegularExpression>(msg) << "\n"; \
+        FAIL() << errMsg.toStdString();                                                            \
+    } else                                                                                         \
+        logMessagesExpected.clear();
+
+#define EXPECT_LOG_MSG(type, exp) \
+    logMessagesExpected.push_back(std::make_tuple(type, QRegularExpression(exp)))
+
+extern QList<std::tuple<QtMsgType, QRegularExpression>> logMessagesExpected;
+void logCapture(QtMsgType msgType, const QMessageLogContext&, const QString& msg);
```

Comment:
this is not a good testing strategy IMO (unless you convince me otherwise; I don't have a giant amount of testing experience). From what I can tell, this is mainly used for inspecting the failure handling during parsing. So this is really a codesmell that indicates the sub-par parsing logic. I think we can postpone that for now, but I feel like a proper way of parsing would involve inspecting the generated `ScreenInfo` struct and checking all the fields to contain the "noValue-value" (so `kMaxMsaa` for example, or whatever value is the default, or wrap each member in an an `optional` (we don't have 1000 `ScreenInfo`s, so I don't think the wasted space due to alignment is an issue).

Reply:
Since logging an error message is the way to inform the user that there was a problem parsing the mapping, yes it is a good strategy IMO. 
Please note that the `ScreenInfo` is already being tested as well, thanks to the `addScreenInfo` mock

Reply:
> we testing whether the exact wording of the log message is what we expect

That's not quite right. We are making sure that a message is delivered to the user.  Using `QRegularExpression`, we make sure a specific pattern is present. This way, if during some refactoring the message is somehow not displayed anymore, we failed the test.
In the future, it could be that the log message is replaced by a dialog messages displayed to the user. These test cases will make sure that when this happen, the developer should take care of asserting the message still get delivered to the user. (not in the log this time obviously)

> Asserting the logmessages is like testing UI by asserting the pixels have the right color IMO.

That's not true. I'm not testing that an exact array of byte is being written to stdout, I'm checking that a specific text pattern with important information to provide to the user is present in an expected receiver (warning logs). This is similar to what you would do with testing framework like Selenium/Cypress/Playwright, where you would make sure a certain component contains a specific message that the user must be communicated with: you wouldn't test pixels have the right color, but you would select a component and ensure its state or content is matching an expected state